patches/net/minecraft/server/jsonrpc/internalapi/MinecraftApi.java.patch

```diff
@@ -0,0 +1,71 @@
+--- a/net/minecraft/server/jsonrpc/internalapi/MinecraftApi.java
++++ b/net/minecraft/server/jsonrpc/internalapi/MinecraftApi.java
+@@ -16,6 +_,22 @@
+     private final MinecraftServerSettingsService minecraftServerSettingsService;
+     private final MinecraftServerStateService minecraftServerStateService;
+     private final MinecraftExecutorService executorService;
++    private final @org.jetbrains.annotations.Nullable DedicatedServer server;
++    private final @org.jetbrains.annotations.Nullable JsonRpcLogger logger;
++
++    public MinecraftApi(
++            NotificationManager p_449655_,
++            MinecraftAllowListService p_443286_,
++            MinecraftBanListService p_443467_,
++            MinecraftPlayerListService p_442735_,
++            MinecraftGameRuleService p_442736_,
++            MinecraftOperatorListService p_443566_,
++            MinecraftServerSettingsService p_443225_,
++            MinecraftServerStateService p_443354_,
++            MinecraftExecutorService p_449378_
++    ) {
++        this(p_449655_, p_443286_, p_443467_, p_442735_, p_442736_, p_443566_, p_443225_, p_443354_, p_449378_, null, null);
++    }
+ 
+     public MinecraftApi(
+         NotificationManager p_449655_,
+@@ -26,7 +_,9 @@
+         MinecraftOperatorListService p_443566_,
+         MinecraftServerSettingsService p_443225_,
+         MinecraftServerStateService p_443354_,
+-        MinecraftExecutorService p_449378_
++        MinecraftExecutorService p_449378_,
++        @org.jetbrains.annotations.Nullable DedicatedServer server,
++        @org.jetbrains.annotations.Nullable JsonRpcLogger logger
+     ) {
+         this.notificationManager = p_449655_;
+         this.allowListService = p_443286_;
+@@ -37,6 +_,8 @@
+         this.minecraftServerSettingsService = p_443225_;
+         this.minecraftServerStateService = p_443354_;
+         this.executorService = p_449378_;
++        this.server = server;
++        this.logger = logger;
+     }
+ 
+     public <V> CompletableFuture<V> submit(Supplier<V> p_442544_) {
+@@ -79,6 +_,14 @@
+         return this.notificationManager;
+     }
+ 
++    public @org.jetbrains.annotations.Nullable DedicatedServer getServer() {
```

Comment:
Instead of nullifying these methods (which would move a burden of null checking to each and every single registered rcp method), have them throw if the server is null. It's already unlikely that someone else will construct a `MinecraftApi` instance and if they do, they should be using our constructor.